api/sparse_host_list_system_profile.py

```diff
@@ -97,6 +98,10 @@ def get_sparse_system_profile(host_id_list, page, per_page, order_by, order_how,
 
     if fields.get("system_profile"):
         variables["fields"] = list(fields.get("system_profile").keys())
+        system_profile_schema = process_system_profile_spec()
+        for field in variables["fields"]:
+            if field not in system_profile_schema.keys():
+                flask.abort(400, "Requested system_profile field is not present in the schema.")
```

Comment:
A bit more detail could be useful here. Something like this:
```suggestion
                flask.abort(400, f"Requested field '{field}' is not present in the system_profile schema.")
```